vite.config.ts

```diff
@@ -0,0 +1,103 @@
+            "@ourworldindata/grapher": "@ourworldindata/grapher/src/index.ts",
+            "@ourworldindata/utils": "@ourworldindata/utils/src/index.ts",
+            "@ourworldindata/core-table":
+                "@ourworldindata/core-table/src/index.ts",
+        },
+    },
+    define: {
+        // Replace all clientSettings with their respective values, i.e. assign e.g. BUGNSAG_API_KEY to process.env.BUGNSAG_API_KEY
+        // it's important to note that we only expose values that are present in the clientSettings file - not any other things that are stored in .env
+        ...Object.fromEntries(
+            Object.entries(clientSettings).map(([key, value]) => [
+                `process.env.${key}`,
+                JSON.stringify(value),
+            ])
+        ),
+    },
+    build: {
+        manifest: true, // creates a manifest.json file, which we use to determine which files to load in prod
+        emptyOutDir: true,
+        outDir: "dist",
+        sourcemap: true,
+        cssCodeSplit: true,
+        rollupOptions: {
+            cache: false, // https://github.com/vitejs/vite/issues/2433#issuecomment-1361094727
+            input: {
+                owid: "./site/owid.entry.ts",
+                admin: "./adminSiteClient/admin.entry.ts",
+            },
+            output: {
+                /**
+                 * This is a bit brittle tbh.
+                 * We want our JS files to be named owid.mjs, common.mjs, and admin.mjs,
+                 * and the CSS assets to be named common.css, owid.css, and admin.css.
+                 *
+                 * Actually, some slight variations of that would also be fine, for example not having the split into common and owid file.
+                 *
+                 * This seems to be the only to do this for now.
+                 */
+                assetFileNames: (assetInfo) => {
+                    if (assetInfo.name?.endsWith(".css")) {
+                        if (assetInfo.name.includes("admin")) {
+                            return "assets/admin.css"
+                        } else if (assetInfo.name.includes("owid")) {
+                            return "assets/owid.css"
+                        } else {
+                            return "assets/common.css"
+                        }
+                    }
+                    return "assets/[name]-[hash][extname]"
+                },
+                chunkFileNames() {
+                    // there's only one chunk currently, so we can do this
+                    return "assets/common.mjs"
+                },
+                entryFileNames(entryInfo) {
+                    if (entryInfo.name === "admin") {
+                        return "assets/admin.mjs"
+                    } else if (entryInfo.name === "owid") {
+                        return "assets/owid.mjs"
+                    }
+                    return "assets/[name]-[hash][extname]"
+                },
```

Comment:
This is, erm, not great.
This is mostly rollup's fault (which is the tool vite is using for production builds), but... sometimes being a little more configurable goes a long way. Oh well.